repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -268,13 +268,8 @@ class ReposeRoutingServlet @Inject()(@Value(ReposeSpringProperties.CORE.REPOSE_V
     }
 
     // Translate the HTTP client response to a servlet response
-    val responseCode = clientResponse.getStatusLine.getStatusCode
-    val responseProcessor = new HttpComponentResponseProcessor(clientResponse, servletResponse, responseCode)
-    if (responseCode >= HttpServletResponse.SC_MULTIPLE_CHOICES && responseCode < HttpServletResponse.SC_NOT_MODIFIED) {
-      responseProcessor.sendTranslatedRedirect()
-    } else {
-      responseProcessor.process()
-    }
+    val responseProcessor = new HttpComponentResponseProcessor(clientResponse, servletResponse)
+    responseProcessor.process()
```

Comment:
This feels kind of stupid. why are we making a thing, and then calling a no argument method on it to mutate something we passed in the constructor?

Reply:
In this case, I just didn't change it. That's how we have been doing it.

When I was rewriting parts of the routing code, I strongly considered rewriting the request and response processors to be collections of static functions, I just never went ahead and did it.

Of course, internal to the processors being an object means that processing can be split across multiple methods with shared state rather than requiring that shared state to be passed around as arguments. Which is not a good reason to continue doing things that way, it's just an observation, mostly.

Reply:
That was all to say, I agree, but my question is whether to make that change as part of this PR, or leave it for a different story?

Reply:
you already changed basically every line in the class being called. Just go ahead and remove the stupid, on this one at least.

Reply:
Done.